File: 1-patrons-de-creations/abstract-factory/code-java/app/Application.java
package app;

import factories.abstract_factories.GuiFactory;
import gui_elements.interfaces.*;

public class Application {

    private GuiFactory factory;
    private Button button;
    private Checkbox checkbox;

    public Application(GuiFactory factory) {

        this.factory = factory;
    };

    public void createUiKids() {
        this.button = factory.createButton();
        this.checkbox = factory.createCheckbox();
    }

    public void paint() {
        this.button.paint();
        this.checkbox.paint();
    }
}

File: 1-patrons-de-creations/abstract-factory/code-typescript/application.ts
import { GUIFactory } from "./factories/abstractFactories/guiFactory.js";
import { Button } from "./guiElements/interfaces/button.js";
import { Checkbox } from "./guiElements/interfaces/checkbox.js";

export class Application {
  private button: Button;
  private checkbox: Checkbox;
  constructor(private factory: GUIFactory) {}

  createUiKid() {
    this.button = this.factory.createButton();
    this.checkbox = this.factory.createCheckbox();
  }

  pain(): void {
    this.button.paint();
    this.checkbox.paint();
  }
}
